fix: refactor-scan 워크플로 LiteLLM 호출 시 ARG_MAX 초과로 실패

### 파트
be

### 증상
`Refactor Scan` 워크플로(`.github/workflows/refactor-scan.yml`)가 dev 브랜치에 push될 때마다 실패함. `Call LiteLLM` 스텝에서 다음 에러로 종료:

```
scripts/refactor-scan/call_litellm.sh: line 30: /usr/bin/jq: Argument list too long
##[error]Process completed with exit code 126.
```

dev에 push되는 거의 모든 커밋(매트릭스 fe/be/ai 전부)에서 동일하게 재현되고 있어서, 자동 리팩토링 스캔이 한참 동안 돌지 않은 상태.

### 원인 (코드 확인 결과)

`scripts/refactor-scan/call_litellm.sh`

1. **line 20-30**: `jq -n --arg content "$prompt" ...`
   - `--arg`는 값을 jq의 커맨드라인 인자로 전달. refactor-scan은 fe/be/ai의 소스 컨텍스트(`collect_context.sh`로 모은 diff/파일 내용)와 프롬프트 본문을 합쳐 표준입력으로 받기 때문에 prompt 크기가 큼. 리눅스 `ARG_MAX`(보통 약 128KB) 한계를 넘어가서 `E2BIG`로 죽음.

2. **line 66 부근**: `curl --data "$payload" ...`
   - 동일한 이유로 잠재적 위험. payload는 prompt를 품으니 인자 크기 한계 그대로 노출됨. 지금은 jq 단계에서 먼저 죽어서 발현은 안 됐지만 fix 후 payload가 살아남으면 동일 에러 가능.

### 재현 방법
1. dev 브랜치에 `apps/be/**`, `apps/fe/**`, `apps/ai/**` 경로 변경을 포함한 커밋을 push
2. `Refactor Scan` 워크플로의 `scan (fe)` / `scan (be)` / `scan (ai)` job 중 임계치 넘은 part에서 `Call LiteLLM` 스텝 실패 확인
3. 가장 최근 실패 run: https://github.com/26-ewha-capstone-logue/logue/actions/runs/26331556272

### 기대 동작
- prompt 크기가 커도 LiteLLM 호출 성공
- LLM 응답을 받아 `Create issue` 스텝까지 정상 도달
- `LAST_SCAN_SHA_{FE,BE,AI}` 변수도 정상 갱신

### 환경
- OS: ubuntu-24.04 (GitHub Actions runner)
- 브랜치/커밋: `dev` 최신
- jq, curl 기본 설치 버전

### 로그/스크린샷
```
scan (fe)  Call LiteLLM  LITELLM_BASE_URL: http://43.201.226.184:4000
scan (fe)  Call LiteLLM  LITELLM_MODEL: claude-sonnet-4-6
scan (fe)  Call LiteLLM  scripts/refactor-scan/call_litellm.sh: line 30: /usr/bin/jq: Argument list too long
scan (fe)  Call LiteLLM  ##[error]Process completed with exit code 126.
```

### 작업 항목

- [ ] **`call_litellm.sh` 리팩터** (`scripts/refactor-scan/call_litellm.sh`)
  1. `tmpdir` 생성을 스크립트 윗부분으로 옮기고 prompt/payload 모두 같은 디렉터리에 둠
  2. stdin을 변수 대신 `$tmpdir/prompt` 파일로 받음 (`cat > "$prompt_file"`)
  3. `jq -n` 호출에서 `--arg content "$prompt"` → `--rawfile content "$prompt_file"` 로 교체. payload도 `--output > "$payload_file"` 로 파일로 떨굼
  4. `curl --data "$payload"` → `curl --data-binary @"$payload_file"` 로 교체 (`curl`의 ARG_MAX 한계 회피)
  5. 기존 trap 정리 로직 통합

### 검증 계획
- 로컬에서 큰 사이즈의 더미 prompt (예: 200KB 이상)를 stdin으로 흘려서 스크립트가 죽지 않는지 확인
- `workflow_dispatch`로 `Refactor Scan` 수동 실행 → `Call LiteLLM` 통과 + `Upload artifacts`에 `/tmp/llm.md` 정상 생성 확인
- dev에 다음 push 발생 시 자동 매트릭스 잡이 success로 끝나는지 모니터

### 리스크
- 거의 없음. 같은 데이터를 인자 → 파일로 전달 방식만 바꾸는 변경. LiteLLM API 호출 페이로드 내용 자체는 동일.
- `--data-binary`는 `--data`와 달리 개행 변환을 안 함. JSON payload라 영향 없음.

--- a/scripts/refactor-scan/call_litellm.sh
+++ b/scripts/refactor-scan/call_litellm.sh
@@ -10,24 +10,29 @@ if ! command -v jq >/dev/null 2>&1; then
   exit 2
 fi
 
-prompt="$(cat)"
-if [[ -z "$prompt" ]]; then
+tmpdir="$(mktemp -d)"
+trap 'rm -rf "$tmpdir"' EXIT
+prompt_file="$tmpdir/prompt"
+payload_file="$tmpdir/payload"
+body_file="$tmpdir/body"
+status_file="$tmpdir/status"
+
+cat > "$prompt_file"
+if [[ ! -s "$prompt_file" ]]; then
   echo "error: empty prompt on stdin" >&2
   exit 2
 fi
 
-payload="$(
-  jq -n \
-    --arg model "$LITELLM_MODEL" \
-    --arg content "$prompt" \
-    --argjson max_tokens "$LITELLM_MAX_TOKENS" \
-    '{
-      model: $model,
-      max_tokens: $max_tokens,
-      temperature: 0.2,
-      messages: [{role: "user", content: $content}]
-    }'
-)"
+jq -n \
+  --arg model "$LITELLM_MODEL" \
+  --rawfile content "$prompt_file" \
+  --argjson max_tokens "$LITELLM_MAX_TOKENS" \
+  '{
+    model: $model,
+    max_tokens: $max_tokens,
+    temperature: 0.2,
+    messages: [{role: "user", content: $content}]
+  }' > "$payload_file"
 
 endpoint="${LITELLM_BASE_URL%/}/v1/chat/completions"
 
@@ -45,11 +50,6 @@ if [[ ${#keys[@]} -eq 0 ]]; then
   exit 2
 fi
 
-tmpdir="$(mktemp -d)"
-trap 'rm -rf "$tmpdir"' EXIT
-body_file="$tmpdir/body"
-status_file="$tmpdir/status"
-
 call_once() {
   local key="$1"
   : >"$body_file"
@@ -63,7 +63,7 @@ call_once() {
       --write-out '%{http_code}' \
       --header "Authorization: Bearer $key" \
       --header 'Content-Type: application/json' \
-      --data "$payload" \
+      --data-binary @"$payload_file" \
       "$endpoint" 2>"$tmpdir/curl_err"
     echo "__exit:$?"
   )"

